utils/project.go

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package utils
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"strings"
+	"github.com/eclipse/codewind-installer/errors"
+)
+
+// CWSettings represents the .cw-settings file which is written to a project
+type CWSettings struct {
+	ContextRoot       string   `json:"contextRoot"`
+	InternalPort      string   `json:"internalPort"`
+	HealthCheck       string   `json:"healthCheck"`
+	InternalDebugPort *string  `json:"internalDebugPort"`
+	IgnoredPaths      []string `json:"ignoredPaths"`
+	MavenProfiles     []string `json:"mavenProfiles,omitempty"`
+	MavenProperties   []string `json:"mavenProperties,omitempty"`
+}
+
+// DetermineProjectInfo returns the language and build-type of a project
+func DetermineProjectInfo(projectPath string) (string, string) {
+	language, buildType := "unknown", "docker"
+	if _, err := os.Stat(path.Join(projectPath, "pom.xml")); err == nil {
+		language = "java"
+		buildType = determineJavaBuildType(projectPath)
+	}
+	if _, err := os.Stat(path.Join(projectPath, "package.json")); err == nil {
+		language = "nodejs"
+		buildType = "nodejs"
+	}
+	if _, err := os.Stat(path.Join(projectPath, "Package.swift")); err == nil {
+		language = "swift"
+		buildType = "swift"
+	}
+	return language, buildType
+}
+
+// CheckProjectPath will stop the process and return an error if path does not
+// exist or is invalid
+func CheckProjectPath(projectPath string) {
+	if projectPath == "" {
+		log.Fatal("Project path has not been set")
```

Comment:
comment: this function does 2 things - it validates the project path, and controls logic flow based on the result (i.e. if the path is invalid, this will call `log.Fatal`, which calls `os.exit`). If we ever want to `CheckProjectPath` without risking an `os.exit`, then we should separate the two. 

(This is a general point about our error handling in this CLI - a lot of functions throw errors rather than returning them, which means that if in future want to handle errors we will have to adapt our code. See https://stackoverflow.com/questions/33885235/should-a-go-package-ever-use-log-fatal-and-when )